uitools/examples/cpp_quick/src/demos/PopupViewDemoModel.cpp

```diff
@@ -0,0 +1,119 @@
+#include "PopupViewDemoModel.h"
+
+#include "GeoView.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "PopupManager.h"
+#include "Scene.h"
+#include "SceneQuickView.h"
+#include "ServiceFeatureTable.h"
+
+#include <QList>
+
+using namespace Esri::ArcGISRuntime;
+
+PopupViewDemoModel::PopupViewDemoModel(QObject* parent) :
+  BaseDemoModel(parent)
+{
+  connect(this, &BaseDemoModel::geoViewChanged, this, &PopupViewDemoModel::setUp);
+}
+
+PopupViewDemoModel::~PopupViewDemoModel()
+{
+}
+
+Esri::ArcGISRuntime::Map* PopupViewDemoModel::initMap_(QObject* parent) const
+{
+  return new Map(QUrl("https://runtime.maps.arcgis.com/home/webmap/"
+                      "viewer.html?webmap=e4c6eb667e6c43b896691f10cc2f1580"),
+                 parent);
+}
+
+Scene* PopupViewDemoModel::initScene_(QObject* parent) const
+{
+  Scene* scene = BaseDemoModel::initScene_(parent);
+  Viewpoint viewPoint(Envelope(-122.5277, 37.7204, -122.3511, 37.7956, SpatialReference(4326)));
+  scene->setInitialViewpoint(viewPoint);
+  FeatureLayer* fl = new FeatureLayer(new ServiceFeatureTable(
+                                          QUrl("https://sampleserver6.arcgisonline.com/arcgis/rest/services/"
+                                               "SF311/FeatureServer/0"),
+                                          parent),
+                                      parent);
+  scene->operationalLayers()->append(fl);
+  return scene;
+}
+
+QObject* PopupViewDemoModel::popupManager_()
+{
+  return m_popupManager;
+}
+
+void PopupViewDemoModel::setPopupManager_(QObject* popupManager)
+{
+  if (popupManager == m_popupManager)
+    return;
+
+  auto oldManager = m_popupManager;
+  m_popupManager = popupManager;
+  emit popupManagerChanged();
+  delete oldManager;
+}
+
+void PopupViewDemoModel::setUp()
+{
+  apply([this](auto geoView)
+        {
+          using ViewType = std::remove_pointer_t<decltype(geoView)>;
+          connect(geoView, &ViewType::mouseClicked, this, [this, geoView](QMouseEvent& mouse)
+                  {
+                    m_featureLayer = static_cast<FeatureLayer*>(geoModel()->operationalLayers()->first());
```

Comment:
you can check the `LayerType` of the feature layer before static casting. Looks like this sample uses an online web map so if someone was to change that map this will crash